### PR TITLE
Add Graviton / arm64 + linux runners into CI

### PR DESCRIPTION
**Pull request by imdaisyadams**

### CODE REVIEW

File: .github/actions/sccache/action.yml [hunk 1]
```diff
@@ -5,22 +5,30 @@ runs:
...
     - if: runner.os == 'macOS' && runner.arch == 'ARM64'
       run: |
         echo "SCCACHE_ARCH=aarch64-apple-darwin" >> $GITHUB_ENV
+        echo "SCCACHE_VERSION=v0.3.3" >> $GITHUB_ENV
+      shell: bash
+
+    - if: runner.os == 'Linux' &&  runner.arch == 'ARM64'
+      run: |
+        echo "SCCACHE_ARCH=aarch64-unknown-linux-musl" >> $GITHUB_ENV
+        echo "SCCACHE_VERSION=v0.4.0-pre.5" >> $GITHUB_ENV
```
Comment (mothran) [inline .github/actions/sccache/action.yml:27]: Is there a published version that is the same between all versions of sccache? I worry that mixed versions on different hosts could cause weird random failures.

**Review by mothran — CHANGES_REQUESTED** [review]

File: .github/actions/sccache/action.yml (hunk 1 above)
Comment (imdaisyadams) [reply .github/actions/sccache/action.yml]: just updated sccache to v.0.5.0 across all systems, and builds are completing successfully

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -59,6 +59,9 @@ jobs:
           - os: macOS
             feature: metal
             device: apple_m2_pro
+          - os: Linux
+            feature: cuda
+            device: graviton2
```
Comment (flaub) [inline .github/workflows/main.yml:64]: Wouldn't the device be `nvidia_rtx_????` here?

Comment (flaub) [reply .github/workflows/main.yml:64]: AFAIK, graviton represents the CPU, but we still have NVidia gear that we're running accelerated code on

Comment (imdaisyadams) [reply .github/workflows/main.yml]: yeah you're right, I've updated the tag to be: `nvidia_tesla_t4g`

**Review by flaub — APPROVED** [review]

File: .github/actions/sccache/action.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ runs:
   steps:
     - if: runner.os == 'Linux' && runner.arch == 'X64'
       run: |
-        echo "SCCACHE_ARCH=x86_64-unknown-linux-musl" >> $GITHUB_ENV
+          echo "SCCACHE_ARCH=x86_64-unknown-linux-musl" >> $GITHUB_ENV
```
Comment (mothran) [inline .github/actions/sccache/action.yml:8]: Extra indent?

**Review by mothran — APPROVED** [review]
Just one little indent change but looks good!

**Pull request merged**